oit-dual-depth-peeling.html

```diff
@@ -0,0 +1,722 @@
+        <div id="features">
+            Features: Vertex Arrays, Uniform Buffers, Immutable Textures, Multiple Render Targets, Float Textures, texelFetch, Hardware Instancing, EXT_color_buffer_float
+        </div>
+        <div>
+            <a href="https://github.com/tsherif/webgl2examples/blob/master/oit.html">Source code</a>
+        </div>
+    </div>
+    <canvas id="gl-canvas"></canvas>
+    <script type="x-shader/vs" id="vertex-peeling">
+        #version 300 es
+
+        layout(std140, column_major) uniform;
+        
+        layout(location=0) in vec4 position;
+        layout(location=1) in vec2 uv;
+        layout(location=2) in vec3 normal;
+        layout(location=3) in vec4 color;
+        layout(location=4) in mat4 modelMatrix;
+        
+        uniform SceneUniforms {
+            mat4 uViewProj;
+            vec4 uEyePosition;
+            vec4 uLightPosition;
+        };       
+        
+        out vec3 vPosition;
+        out vec2 vUV;
+        out vec3 vNormal;
+        flat out vec4 vColor;
+
+        void main() {
+            vec4 worldPosition = modelMatrix * position;
+            vPosition = worldPosition.xyz;
+            vUV = uv;
+            vNormal = mat3(modelMatrix) * normal;
+            vColor = color;
+            gl_Position = uViewProj * worldPosition;
+        }
+    </script>
+    <script type="x-shader/fs" id="fragment-peeling">
+        #version 300 es
+        precision highp float;
+        precision highp sampler2D;
+
+        #define MAX_DEPTH 99999.0
+
+        layout(std140, column_major) uniform;
+
+        uniform SceneUniforms {
+            mat4 uViewProj;
+            vec4 uEyePosition;
+            vec4 uLightPosition;
+        };
+
+        uniform sampler2D uTexture;
+        uniform sampler2D uDepth;
+        uniform sampler2D uFrontColor;
+        
+        in vec3 vPosition;
+        in vec2 vUV;
+        in vec3 vNormal;
+        flat in vec4 vColor;
+
+        layout(location=0) out vec4 depth;  // RG32F, R - negative front depth, G - back depth
+        layout(location=1) out vec4 frontColor;
+        layout(location=2) out vec4 backColor;
+
+        void main() {
+
+            // -------------------------
+            // dual depth peeling
+            // -------------------------
+
+            float fragDepth = gl_FragCoord.z;   // 0 - 1
+
+            ivec2 fragCoord = ivec2(gl_FragCoord.xy);
+            vec2 lastDepth = texelFetch(uDepth, fragCoord, 0).rg;
+            vec4 lastFrontColor = texelFetch(uFrontColor, fragCoord, 0);
+
+            // depth value always increases
+            // so we can use MAX blender equation
+            depth.rg = vec2(-MAX_DEPTH);
+
+            // front color always increases
+            // so we can use MAX blender equation
+            frontColor = lastFrontColor;
+
+            // back color is separatly blend afterwards each pass
+            backColor = vec4(0.f);

```

Comment:
Just to follow the conventions set up in the rest of the code of writing out floats with trailing 0, no 'f', make this `0.0`. Same for instances below.